core/src/main/java/org/elasticsearch/action/search/SearchScrollQueryAndFetchAsyncAction.java

```diff
@@ -170,7 +171,7 @@ private void finishHim() {
     private void innerFinishHim() throws Exception {
         ScoreDoc[] sortedShardList = searchPhaseController.sortDocs(true, queryFetchResults);
         final InternalSearchResponse internalResponse = searchPhaseController.merge(sortedShardList, queryFetchResults,
-            queryFetchResults);
+                queryFetchResults, null);
```

Comment:
Instead of using null here to disable the reduction of aggregations, I'm wondering that we should always pass the search request, but then in `searchPhaseController.merge`, we would only reduce if shards returned a non-empty aggregation response?